multicluster/client.go

```diff
@@ -99,11 +99,19 @@ func (m *gatedClient) RESTMapper() meta.RESTMapper {
 	return m.base.RESTMapper()
 }
 
-func (m *gatedStatusWriter) Update(ctx context.Context, obj client.Object, opts ...client.UpdateOption) error {
+func (m *gatedClient) SubResource(subResource string) client.SubResourceClient {
+	return m.base.SubResource(subResource)
```

Comment:
The implementation here is incorrect.

Reply:
SubResource is extra, and did not call

Reply:
But still it's not a good idea to put a random implementation here.

Reply:
<img width="1411" alt="image" src="https://user-images.githubusercontent.com/18587688/219536484-b0e0aab8-7865-4203-a99f-2744cffff0d0.png">
The source code is placed after the Status method

Reply:
SubResource client is used for subresource requests so for the gated multicluster client, we also need a gated subresource multicluster client and here you should call the delegated client.